javaServices/coreJavaServices/src/joliex/util/XmlUtils.java

```diff
@@ -158,6 +183,43 @@ public Value xmlToValue( Value request )
 			documentBuilderFactory.setSchema(null); // reset schema
 		}
 	}
+	
+	public Value xmlToPlainValue( Value request )
+		throws FaultException
+	{
+		try {
+			
+			Value result = Value.create();
+			InputSource src;
+			if ( request.isByteArray() ) {
+				src = new InputSource( new ByteArrayInputStream( request.byteArrayValue().getBytes() ) );
+			} else {
+				src = new InputSource( new StringReader( request.strValue() ) );
+			}
+			
+			if ( request.getFirstChild( "options" ).hasChildren( "charset" ) ) {
+				src.setEncoding( request.getFirstChild( "options" ).getFirstChild( "charset" ).strValue() );
+			}
+		
+			DocumentBuilder builder = documentBuilderFactory.newDocumentBuilder();
+			Document doc = builder.parse( src );
+			jolie.xml.XmlUtils.documentToPlainValue( doc, result );
+			
+			return result;
+		} catch( ParserConfigurationException e ) {
+			e.printStackTrace();
+			throw new FaultException( e );
+		} catch( SAXException e ) {
+			e.printStackTrace();
+			throw new FaultException( e );
+		} catch( IOException e ) {
+			e.printStackTrace();
+			throw new FaultException( e );
+		} finally {
+			documentBuilderFactory.setSchema(null); // reset schema
```

Comment:
This seems to indicate that this entire service cannot be accessed concurrently. Am I right? If so, this should be fixed.